tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Inheritance.java

```diff
@@ -0,0 +1,149 @@
+            value = {
+                @APIResponse(
+                        responseCode = "200",
+                        description = "Array of possible alt values",
+                        content =
+                                @Content(
+                                        mediaType = "application/json",
+                                        schema = @Schema(implementation = Map.class)))
+            })
+    public Response getReasons() {
+        Map<String, String> map = new TreeMap<>();
+        for (final LocaleInheritanceInfo.Reason r : LocaleInheritanceInfo.Reason.values()) {
+            map.put(r.name(), r.getDescription());
+        }
+        return Response.ok(map).build();
+    }
+
+    @GET
+    @Path("/locale/{localeId}/{hexId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(
+            summary = "Explain inheritance of an xpath",
+            description = "Looks up the xpath and explains its inheritance")
+    @APIResponses(
+            value = {
+                @APIResponse(
+                        responseCode = "404",
+                        description = "XPath Hex ID not found",
+                        content =
+                                @Content(
+                                        mediaType = "application/json",
+                                        schema = @Schema(implementation = STError.class))),
+                @APIResponse(
+                        responseCode = "200",
+                        description = "Array of possible alt values",
+                        content =
+                                @Content(
+                                        mediaType = "application/json",
+                                        schema =
+                                                @Schema(
+                                                        implementation =
+                                                                InheritanceResponse.class))),
+            })
+    public Response getInheritanceByHex(
+            @Parameter(required = true, example = "aa", schema = @Schema(type = SchemaType.STRING))
+                    @PathParam("localeId")
+                    String localeId,
+            @Parameter(
+                            required = true,
+                            example = "6154e7673c3829ce",
+                            schema = @Schema(type = SchemaType.STRING))
+                    @PathParam("hexId")
+                    String hexId) {
+        CLDRLocale locale = CLDRLocale.getInstance(localeId);
+        if (locale == null) {
+            return STError.badLocale(localeId);
+        }
+        String xpath = getXPathByHex(hexId);
+        if (xpath == null) {
+            return STError.badPath(hexId);
+        }
+
+        final CLDRFile f = sm.getSTFactory().make(locale, true);
+        final List<LocaleInheritanceInfo> paths = f.getPathsWhereFound(xpath);
+        return Response.ok(new InheritanceResponse(paths)).build();
```

Comment:
if there's the slightest chance that `new InheritanceResponse(paths))` could throw an exception, it would be best to call it in a try block and return 500 if it throws

safest not to call ok until it's confirmed ok

Reply:
Throwing here should throw a 500 automatically

Reply:
what we've seen in other similar situations, is that the response is mangled

better something like this:

```
         try {
             InheritanceResponse response = new InheritanceResponse(paths)
             return Response.ok(response).build();
         } catch (Exception e) {
             return Response.status(500, "An exception occurred").entity(e).build();
         }
```

See https://unicode-org.atlassian.net/browse/CLDR-16223 which BTW is assigned to you as major for v44, but still "new"?

Reply:
by "mangled" I mean for example "Error: could not load: Status 200 OK; URL: /cldr-apps/api/voting/reports/locales/sv/reports/personnames.html?&cacheKill=1670013698549 Missing response; responseType = text" -- as reported in https://unicode-org.atlassian.net/browse/CLDR-16205

Reply:
> what we've seen in other similar situations, is that the response is mangled
> 
> 
> 
> better something like this:
> 
> 
> 
> ```
> 
>          try {
> 
>              InheritanceResponse response = new InheritanceResponse(paths)
> 
>              return Response.ok(response).build();
> 
>          } catch (Exception e) {
> 
>              return Response.status(500, "An exception occurred").entity(e).build();
> 
>          }
> 
> ```
> 
> 
> 
> See https://unicode-org.atlassian.net/browse/CLDR-16223 which BTW is assigned to you as major for v44, but still "new"?

The error handler is supposed to do exactly that already for all paths. May be a bug.

Reply:
"may be a bug" -- could be; anyway safest to create a 500 response explicitly

Reply:
So i tried this and any exception does create a 500 automatically, with a stack trace (instead of just 'an exception occurred) and also error logging. I'm not sure what "200 OK" above is, but sounds like a different bug.

Error handling is great if we're going to actually do something with it.  

What might be better instead is to define an ExceptionMapper across our APIs… that uses a standardized response (we can use STError) to return something like `{"err": "An exception occurred"}` with the 500 instead of the HTML page.   Something like [this](https://www.ibm.com/docs/en/was-nd/8.5.5?topic=applications-defining-exception-mappers-resource-exceptions-errors) would do that.  Otherwise, `Response.status(500, "An exception occurred").entity(e).build()` returns less information than the standard mapper and doesn't log. 

<img width="472" alt="image" src="https://user-images.githubusercontent.com/855219/233674893-6315137b-adc0-4bbf-9a0e-d9810e25e03b.png">

Reply:
I guess it depends on what exactly goes on in the response-returning code, whether an exception there results in a well-formed and informative 500 response, or a bogus 200 response, as in the old person-names report, when it looked like this:

```
  return Response.ok(new StreamingOutput() {
        @Override
        public void write(OutputStream output) throws IOException, WebApplicationException {
            if (chart != null) {
                chart.writeContents(output, stf);
            } else {
                switch (report) {
                // "Old Three" reports
                case compact:
                case datetime:
                case zones:
                    try (final Writer w = new OutputStreamWriter(output);) {
                        SurveyAjax.generateReport("r_" + report.name(), w, sm, loc);
                    }
                    break;
                default:
                    try (final Writer w = new OutputStreamWriter(output);) {
                        w.write("Could not load report: " + report.name());
                    }
                }
            }
        }
    }).build();
```

All I know is that was a pain to debug, and I'd feel less apprehensive if we only called `Response.ok(response)` after successfully obtaining the response object.

Reply:
> All I know is that was a pain to debug, and I'd feel less apprehensive if we only called Response.ok(response) after successfully obtaining the response object.

if there's an exception then `Response.ok` isn't called because the exception breaks out.

Reply:
i added comments to CLDR-16205 let's discuss there